### PR TITLE
update powsybl-core and powsybl-diagram versions in pom files

### PR DESCRIPTION
**Pull request by So-Fras**
Signed-off-by: Sophie Frasnedo <sophie.frasnedo@rte-france.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
update versions of powsybl-core and powsybl-diagram in pom files

### CODE REVIEW

File: network-area-diagram-viewer/pom.xml [hunk 1]
```diff
@@ -42,8 +42,8 @@
         <junit.version>4.13.1</junit.version>
         <maven.compiler.source>17</maven.compiler.source>
         <maven.compiler.target>17</maven.compiler.target>
-        <powsybl.diagram.version>3.1.0-SNAPSHOT</powsybl.diagram.version>
-        <powsybl.core.version>5.0.0</powsybl.core.version>
+        <powsybl.diagram.version>3.1.0</powsybl.diagram.version>
```
Comment (flo-dup) [inline network-area-diagram-viewer/pom.xml:45]:
```suggestion
        <powsybl.diagram.version>3.2.0-SNAPSHOT</powsybl.diagram.version>
```

The primary goal of these viewers is to be used with latest snapshot version of tested artifact, so you shouldn't refer to a release.

File: power-factory-viewer/pom.xml [hunk 2]
```diff
@@ -32,7 +32,7 @@
         <javafx.version>11.0.2</javafx.version>
         <slf4j.version>1.7.28</slf4j.version>
 
-        <powsybl-core.version>5.1.0-SNAPSHOT</powsybl-core.version>
+        <powsybl-core.version>5.1.0</powsybl-core.version>
```
Comment (flo-dup) [inline power-factory-viewer/pom.xml:35]:
```suggestion
        <powsybl-core.version>5.2.0-SNAPSHOT</powsybl-core.version>
```

File: single-line-diagram-viewer/pom.xml [hunk 3]
```diff
@@ -42,8 +42,8 @@
         <javafx-mvn-plugin.version>0.0.8</javafx-mvn-plugin.version>
         <slf4j.version>1.7.22</slf4j.version>
 
-        <powsybl-diagram.version>3.1.0-SNAPSHOT</powsybl-diagram.version>
-        <powsybl-core.version>5.0.0</powsybl-core.version>
+        <powsybl-diagram.version>3.1.0</powsybl-diagram.version>
```
Comment (flo-dup) [inline single-line-diagram-viewer/pom.xml:45]:
```suggestion
        <powsybl-diagram.version>3.2.0-SNAPSHOT</powsybl-diagram.version>
```

**Review by flo-dup — CHANGES_REQUESTED** [review]

**Comment by So-Fras** [comment]
The CI is ok, both versions (powsybl-core and powsybl-diagram) seem ok too. 
@flo-dup do you approve the changes ?

**Review by flo-dup — COMMENTED** [review]
I think that for power-factory-viewer you'd rather use 5.2.0-SNAPSHOT, isn't it @geofjamg ?

**Review by flo-dup — APPROVED** [review]

**Pull request merged**